client-go: update leader election example

kairen opened the pull request:
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the leader election example for help developer know how to implement.

* Move  example from `tools/leaderelection` to `examples/leader-election`.
* Improve example codes.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

NONE
```release-note
none
```

vladimirvivien committed:
CSI Inline Volume - API changes

pbarker committed:
adds dynamic audit e2e test

Pingan2017 committed:
add subcommand kubectl create cronjob

yagonobre committed:
Bump minimum kubeadm supported version to v1.12

pohly committed:
e2e/storage: testing of external storage drivers

It is useful to apply the storage testsuite also to "external" (=
out-of-tree) storage drivers. One way of doing that is setting up a
custom E2E test suite, but that's still quite a bit of work.

An easier alternative is to parameterize the Kubernetes e2e.test
binary at runtime so that it instantiates the testsuite for one or
more drivers. Some parameters have to be provided before starting the
test because they define configuration and capabilities of the driver
and its storage backend that cannot be discovered at runtime. This is
done by populating the DriverDefinition with the content of the file
that the new -storage.testdriver parameters points to.

The universal .yaml and .json decoder from Kubernetes is used. It's
flexible, but has some downsides:
- currently ignores unknown fields (see https://github.com/kubernetes/kubernetes/pull/71589)
- poor error messages when fields have the wrong type

Storage drivers have to be installed in the test cluster before
starting e2e.test. Only tests involving dynamically provisioned
volumes are currently supported.

Commit:
kubeadm-reset: add means to clear the ClusterStatus

Add ResetClusterStatusForNode() that clears a certain
control-plane node's APIEndpoint from the ClusterStatus
key in the kubeadm ConfigMap on "kubeadm reset".

Commit:
remove unneeded println

gnufied committed:
Introduce feature gate for volume expansion

Update CSI library version

Commit:
CSI Inline Volume - Generated files

Commit:
CSI Inline Volume - Kubelet/Driver code impl

Commit:
Add "fieldManager" to flag to PATCH/CREATE/UPDATE

And add a corresponding flag in kubectl (for apply), even though the
value is defaulted in kubectl with "kubectl".

The flag is required for Apply patch-type, and optional for other PATCH,
CREATE and UPDATE (in which case we fallback on the user-agent).

kairen committed:
Update leader election example for client-go

kairen commented:
cc @smarterclayton

smarterclayton commented:
/ok-to-test

kairen committed:
Remove leaderelection example srcs in filegroup

kairen commented:
/test pull-kubernetes-e2e-gce-100-performance

kairen committed:
Add leaderelection example srcs in filegroup

kairen commented:
@smarterclayton Could you review this PR again?

kairen commented:
/assign @smarterclayton

Commit:
adds better logging around retry loops in dynamic audit e2e test

fedebongio commented:
/assign @yliaog

Random-Liu committed:
New pod log directory /var/log/pods/NAMESPACE_NAME_UID.

Signed-off-by: Lantao Liu <lantaol@google.com>

tallclair committed:
Migrate RuntimeClass to internal API

tallclair committed:
Migrate to internal node.k8s.io client

Random-Liu committed:
Include pod logs in the pod ephemeral storage.

Signed-off-by: Lantao Liu <lantaol@google.com>

yujuhong committed:
nodelifecycle controller: reconcile node OS/arch labels

liggitt commented on staging/src/k8s.io/client-go/examples/leader-election/main.go:
```diff
@@ -37,42 +37,62 @@ import (
+	var ttlSeconds int
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.StringVar(&resourceName, "resource-name", "example", "The leader election resource name (Configmap or Endpoint name)")
```
suggest pinning this to the purpose-built lease object, not configmap or endpoints

kairen replied:
ok, I'll update from the suggestion.

liggitt commented on staging/src/k8s.io/client-go/examples/leader-election/main.go:
```diff
@@ -37,42 +37,62 @@ import (
 	// Endpoints lock type, where quite a few system agents like the kube-proxy
 	// and ingress controllers must watch endpoints).
-	id := args[0]
 	lock := &resourcelock.ConfigMapLock{
```
suggest changing to LeaseLock

liggitt commented on staging/src/k8s.io/client-go/examples/leader-election/main.go:
```diff
@@ -37,42 +37,62 @@ import (
+
+	flag.StringVar(&kubeconfig, "kubeconfig", "", "absolute path to the kubeconfig file")
+	flag.StringVar(&resourceName, "resource-name", "example", "The leader election resource name (Configmap or Endpoint name)")
+	flag.IntVar(&ttlSeconds, "ttl", 30, "TTL for leader election in seconds")
```
were there issues with the existing 60/15/5 ttl/renew/retry distribution in the example?

yliaog commented on staging/src/k8s.io/client-go/examples/leader-election/main.go:
```diff
@@ -37,42 +37,62 @@ import (
-		log.Fatalf("requires three arguments: ID NAMESPACE CONFIG_MAP_NAME (%d)", len(args))
+
+	podName := os.Getenv("POD_NAME")
+	podNamespace := os.Getenv("POD_NAMESPACE")
```
any particular reason to use environment variable, instead of flags?

kairen replied:
If this example runs in Kubernetes cluster, we can pass Pod name to identify it, but I think this can replace to flags.

davidz627 committed:
GetMountRefs fixed to handle corrupted mounts by treating it like an
unmounted volume

Commit:
adds owner of TODO comments in dynamic audit e2e test

Commit:
fix missing event report handling in dynamic audit e2e

yagonobre committed:
Add certificate-key to kubeadm init upload-certs phase, and improve init output

jingxu97 committed:
Add GCE PD tests for windows cluster

This PR is the first one to add a few GCE PD tests for windows cluster. Will
add more tests in later PRs

tallclair committed:
Add node.k8s.io/v1beta1 API

mtaufen committed:
Rotate windows logs

gnufied committed:
Rename ExandFS to NodeExpand

Handle resize error in online resizing
Use NodeExpandable plugin to mark volumes that require node expansion

pjh committed:
Install Stackdriver agent in Windows test clusters.

msau42 committed:
lock csi and plugin watcher GA feature gates

roycaihw committed:
test client-side validation behavior for CRD

support version-skew testing

davidz627 committed:
Add CSINode initialization for CSIMigration on node startup before pod ready

mtaufen committed:
Update klog dependency

yujuhong committed:
Fix computing of cpu nano core usage

CRI runtimes do not supply cpu nano core usage as it is not part of CRI
stats. However, there are upstream components that still rely on such
stats to function. The previous fix was faulty because the multiple
callers could compete and update the stats, causing
inconsistent/incoherent metrics. This change, instead, creates a
separate call for updating the usage, and rely on eviction manager,
which runs periodically, to trigger the updates. The caveat is that if
eviction manager is completley turned off, no one would compute the
usage.

msau42 committed:
e2e test updates for beta

Commit:
Add CSI migration logic for EBS Volume ID format

gnufied committed:
Move resize function parameters to a new ResizeOptions type

This enables us to pass CSI volume phase

neolit123 committed:
kubeadm: update output of init, join reset commands

- move most unrelated to phases output to klog.V(1)
- rename some prefixes for consistency - e.g.
[kubelet] -> [kubelet-start]
- control-plane-prepare: print details for each generated CP
component manifest.
- uppercase the info text for all "[reset].." lines
- modify the text for one line in reset

roycaihw committed:
apiserver: make endpoint webservice builder helpers public

roycaihw committed:
Makefile.generated_files: generate apiextension-apiserver openapi definition

roycaihw committed:
apiextensions-apiserver: serve openapi spec

Co-authored-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

gnufied committed:
Add  CSI volume resizing tests

Add some tests for checking node expansion
Add new tests for expander

bswartz committed:
Promote CSI raw blocks to beta

sttts committed:
hack/*protobuf.sh: dynamically find api roots

Commit:
removes cluster type restrictions from dynamic audit e2e test

sttts committed:
apis: add +k8s:protobuf-gen=package tag

rjaini committed:
Adding a check to make sure UseInstanceMetadata flag is true to get data from metadata.

brusMX committed:
Creating NodePort test for Windows

Co-authored-by: Muzz Imam <muzzz.imam@gmail.com>

mbohlool committed:
Add AdmissionReviewVersions to admissionregistration and default it

sttts committed:
go-to-protobuf: sort topological by imports

sttts committed:
bump(gonum): same version, but new topo packages

soltysh committed:
Fix panic in kubectl cp command

k8s-ci-robot committed:
Merge pull request #74615 from andrewsykim/delete-pvl-controller

Delete the out-of-tree PV labeler controller

k8s-ci-robot committed:
Merge pull request #74829 from pohly/e2e-ginkgo

test/e2e: fix `ginkgo ./test/e2e`

k8s-ci-robot committed:
Merge pull request #74872 from nikopen/patch-2

cherry picks usage info link

k8s-ci-robot committed:
Merge pull request #74873 from neolit123/fix-join-phase-preflight

kubeadm: add a flag to RunInitNodeChecks to indicate sec. control-plane

k8s-ci-robot committed:
Merge pull request #74882 from feiskyer/convert-rg-to-lower

Convert resource group name in Azure provider ID to lower cases

ipuustin committed:
hack/lib/util.sh: fix empty array expansion with bash 3.

k8s-ci-robot committed:
Merge pull request #74529 from liggitt/kubelet-service-links-error

Kubelet service links error

k8s-ci-robot committed:
Merge pull request #74854 from neolit123/conformance-nodes

conformace/run_e2e.sh: pass shellcheck and autodetect ginkgo nodes

k8s-ci-robot committed:
Merge pull request #74895 from nikhita/publishing-bot-fix

publishing: add component-base as dependency for 1.14 branches

k8s-ci-robot committed:
Merge pull request #74907 from bsalamat/num_cpu

Revert "Use runtime.NumCPU() instead of a fixed value for parallel scheduler threads"

k8s-ci-robot committed:
Merge pull request #74915 from jpbetz/revert-72062-2

revert 72062: Perform GCE master log rotation check every 5 minutes

davidz627 committed:
Add ADC Fallback if Node doesn't have driver installed

k8s-ci-robot committed:
Merge pull request #74868 from qingsenLi/k8s190304

remove some unused const in test/e2e_node

k8s-ci-robot committed:
Merge pull request #74922 from wangzhen127/fix-npd-test-ubuntu

Fix NPD e2e test on clusters with Ubuntu nodes

k8s-ci-robot committed:
Merge pull request #74928 from pjh/gce-windows-reenable-defender

Don't disable Windows Defender.

k8s-ci-robot committed:
Merge pull request #74932 from davidz627/fix/gcePartitionTranslation

Fix 0 partitions defaulting for CSI Translation Library

k8s-ci-robot committed:
Merge pull request #74937 from liggitt/proto-verify

Add proto roundtrip tests in roundtrip_test.go files

liggitt committed:
Scope slow/failing webhook to test namespace

lee0c committed:
Remove dot imports

mbohlool committed:
Add ConversionReviewVersions to CustomResourceDefinitionSpec and defdefault it to v1beta1

mbohlool committed:
Update generated files

k8s-ci-robot committed:
Merge pull request #74222 from Huang-Wei/kubelet-mirrorpod

kubelet: return mirror pod in GetActivePods()

Liujingfang1 committed:
update vendoring of kustomize

tallclair committed:
Promote RuntimeClass to beta

k8s-ci-robot committed:
Merge pull request #74721 from kwiesmueller/sort-fieldmanagers

implement managedFields sorting

k8s-ci-robot committed:
Merge pull request #74830 from msau42/lock-csi-ga

lock csi and plugin watcher GA feature gates

k8s-ci-robot committed:
Merge pull request #74926 from ddebroy/csiprov2

Add new CSI translation APIs required by CSI sidecars and unit tests

k8s-ci-robot committed:
Merge pull request #74933 from yujuhong/fix-cpu-nano-cores

Fix computing of cpu nano core usage

k8s-ci-robot committed:
Merge pull request #74945 from liucimin/fix_e2e_test_typo

Typo fix in e2e test description message: defaut->default

davidz627 committed:
Address review comments

neolit123 committed:
kubeadm: amend flags for join phases

Certain join phases have flags that are redundant:
- getControlPlanePreparePhaseFlags():
  - amend flags (use switch / case)
  - add example for the parent command
  - rename internal function and sub-phase "manifests"
    to "control-plane"
- getControlPlaneJoinPhaseFlags():
  - amend flags
  - remove MacroCommandLongDescription

msau42 committed:
don't check for container status ready in subpath test

Commit:
Disable "should set different fsGroup for second pod if first pod is deleted" test temporarily

dims committed:
Add missing dependency

Change-Id: I4abac0970013d708398c804a97f5e7597eb6b7b0

k8s-ci-robot committed:
Merge pull request #74858 from leakingtapan/ebs-id

Add CSI migration logic for EBS Volume ID format

k8s-ci-robot committed:
Merge pull request #74153 from ixdy/bazel-kubernetes-src-tarball-new

bazel: make kubernetes-src.tar.gz actually include all srcs

k8s-ci-robot committed:
Merge pull request #74477 from liggitt/webhook-admission-scope

Webhook admission scope

k8s-ci-robot committed:
Merge pull request #74612 from jingxu97/windows

Add GcePD windows support

k8s-ci-robot committed:
Merge pull request #74671 from yagonobre/certificate-key

Add certificate-key to kubeadm upload-certs phase, and improve init output

k8s-ci-robot committed:
Merge pull request #74675 from dashpole/cadvisor_godep_v0.33.0

update cadvisor godeps to v0.33.0

danielqsj committed:
clean the deprecated metrics which introduced recently

k8s-ci-robot committed:
Merge pull request #74446 from yanghaichao12/comm0223

Fix typos

tnozicka committed:
Fix client-go fake client example flake

tnozicka committed:
Generated: Update Bazel

k8s-ci-robot committed:
Merge pull request #74655 from bclau/tests/windows-gcrio

tests: Adds gcr.io Windows image test case

k8s-ci-robot committed:
Merge pull request #74743 from lee0c/lecattar/win-mem-limits

Windows nodeAllocatable test

k8s-ci-robot committed:
Merge pull request #74789 from cwdsuzhou/kubectl_invalid

Bugfix: kubectl shows last<invalid> if pv, pod or pvc is terminating using kubectl describe xxxx

k8s-ci-robot committed:
Merge pull request #74870 from neolit123/fix-join-phase

kubeadm: update output of init, join reset commands

k8s-ci-robot committed:
Merge pull request #74905 from stevesloka/numNodesLookup

If num-nodes is not set, then look up via API how many are scheduleable

k8s-ci-robot committed:
Merge pull request #74952 from ping035627/k8s-190305

Fix some typos

k8s-ci-robot committed:
Merge pull request #74959 from deads2k/hook-reg

track poststarthook registration stacks for debugging

k8s-ci-robot committed:
Merge pull request #74964 from bclau/tests/externalname-retry

tests: Adds retry for dig for ExternalName test

k8s-ci-robot committed:
Merge pull request #74970 from ipuustin/lib-util-sh-2

hack/lib/util.sh: fix empty array expansion with bash 3.

k8s-ci-robot committed:
Merge pull request #74981 from jennybuckley/update-vendor

Update sigs.k8s.io/structured-merge-diff version

rosti committed:
kubeadm: Don't hardcode temp path in a test

Hardcoding a temp path of /tmp/... is not portable and can potentially cause
other issues (such as flakyness) too.
Use TempFile instead.

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

k8s-ci-robot committed:
Merge pull request #71950 from bclau/fix-max-backoff-test-flake

tests: getRestartDelay waits for the next Terminated state

k8s-ci-robot committed:
Merge pull request #74969 from tnozicka/fix-example-flake

Fix client-go fake client example flake

k8s-ci-robot committed:
Merge pull request #74989 from liggitt/webhook-test-namespace

Scope slow/failing webhook to test namespace

k8s-ci-robot committed:
Merge pull request #75010 from dims/clean-up-publishing-bot-rules

Cleanup publishing-bot rules

k8s-ci-robot committed:
Merge pull request #75011 from msau42/fix-readonly-e2e

don't check for container status ready in subpath test

k8s-ci-robot committed:
Merge pull request #70265 from jingyih/use_chunking_when_list_in_cronjob_controller

Retrieve list in chunks in cronjob controller

k8s-ci-robot committed:
Merge pull request #75015 from cofyc/fix73168-disable-temporarily

Disable "should set different fsGroup for second pod if first pod is deleted" temporarily

soltysh committed:
Add examples how to run make test-cmd specific tests

errordeveloper committed:
Move printers & resources packages

These packages deserve to live at the top level of `k8s.io/cli-runtime`,
as they are potentially of wider use than genericclioptions.

k8s-ci-robot committed:
Merge pull request #72685 from yagonobre/bump-minimum

Bump minimum kubeadm supported version to v1.13

k8s-ci-robot committed:
Merge pull request #74424 from liggitt/drop-k8s-io-node-labels

Clean up self-set node labels

k8s-ci-robot committed:
Merge pull request #74993 from msau42/csi-beta-e2e

e2e test updates for CSIDriver and CSINode beta

dims committed:
verify script to ensure publishing bot rules and godeps are in sync

Change-Id: Idbad862a797916cc4408ea0099c30d0557044907

k8s-ci-robot committed:
Merge pull request #74787 from jianglingxia/fixbugiscsi

fix bug iscsi volume attach failed of /sys/class/iscsi_host dir not e…

k8s-ci-robot committed:
Merge pull request #75013 from neolit123/fix-join-flags

kubeadm: amend flags for join phases

k8s-ci-robot committed:
Merge pull request #75032 from rosti/win-test

kubeadm: Don't hardcode temp path in a test

k8s-ci-robot committed:
Merge pull request #74737 from wk8/wk8/gmsa_bug_fix

Fixing a small bug with GMSA support

ixdy committed:
bazel: refactor multi-arch container builds into starlark

pjh committed:
Disable Windows Defender in Windows nodes on GCE again.

This reverts commit fbf4fe4714d749ced197ce51b3b806304dda091a. Windows
Defender seems to be causing our Windows nodes to crash and reboot
during e2e tests, e.g.
https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-windows-gce/228/artifacts/e2e-228-36623-windows-node-group-q4.

ixdy committed:
bazel: refactor core docker images to use multi_arch_container

timothysc committed:
Revert "Merge pull request #74628 from hpandeycodeit/alpha_kubeconf"

This reverts commit 7053e43b2e413952fe46f0e9e2afa37b322225a0, reversing
changes made to d58ff3bfc0743320d66ee7778bcf08bf4b1d8038.

ixdy committed:
bazel: add docker_push rules

andrewsykim committed:
fix hack/update-bazel.sh for mac os environments

k8s-ci-robot committed:
Merge pull request #74958 from dims/Promote-mkumatag-to-approver

Promote mkumatag to approver in test/images

k8s-ci-robot committed:
Merge pull request #75037 from soltysh/cp_bug

Fix panic in kubectl cp command

ixdy committed:
bazel maintain support for arch-less-named tarballs

wangzhen127 committed:
Fix NPD e2e test on Ubuntu node and update NPD container version

Commit:
Ignore changes to managed field in noderestriction

The validation is failing because the managedfields are changed when the
object is updated. We don't have a good way to verify that the changes
are only the ones that are supposed to happen, so we'll just ignore them
for now.

grayluck committed:
Allow session affinity a period of time to setup for new services.

This is to deal with the flaky session affinity test.

Commit:
Track dry-run and apply in metrics

tallclair committed:
generated files

k8s-ci-robot committed:
Merge pull request #73840 from adisky/cinder-intree-migration

Add CSI Migration feature flags for Cinder In tree Driver

k8s-ci-robot committed:
Merge pull request #75005 from Liujingfang1/vendor

update vendoring of kustomize

msau42 committed:
make CSI PD tests serial

Commit:
handle validation errors correctly

Commit:
fix handling of nil times in managedFields sorting

roycaihw committed:
apiextensions-apiserver: crd openapi spec builder

feiskyer committed:
Add network stats for Windows containers

wk8 committed:
Adding an e2e test on Windows GMSA support

This patch is adding an e2e on the Windows GMSA support added in
https://github.com/kubernetes/kubernetes/pull/73726 and
https://github.com/kubernetes/kubernetes/pull/74737.

Signed-off-by: Jean Rouge <rougej+github@gmail.com>

feiskyer committed:
Add network stats for node interfaces

feiskyer committed:
Run hack/update-bazel.sh

wk8 committed:
Updating Windows' e2e tests' README to the official repo list

Signed-off-by: Jean Rouge <rougej+github@gmail.com>

k8s-ci-robot committed:
Merge pull request #70036 from pbarker/audit-etoe

dynamic audit e2e test

k8s-ci-robot committed:
Merge pull request #72836 from pohly/storage-test-plugin

e2e/storage: testing of external storage drivers

k8s-ci-robot committed:
Merge pull request #74510 from SataQiu/fix-golint-2019022502

Fix some golint failures for pkg/kubelet/apis/...

k8s-ci-robot committed:
Merge pull request #74748 from pjh/gce-windows-stackdriver

Install Stackdriver agent in Windows nodes on GCE.

k8s-ci-robot committed:
Merge pull request #74807 from runyontr/test-cmd-kubeadm-isolation

Kubeadm test Isolation: #74464

gnufied committed:
Update csi library version

k8s-ci-robot committed:
Merge pull request #74369 from skriss/object-tracker-merge-patch

fake client object tracker: support merge patch

k8s-ci-robot committed:
Merge pull request #74468 from ixdy/bazel-cgo-crossbuild

bazel: improve multi-arch support for docker image artifacts

k8s-ci-robot committed:
Merge pull request #74625 from davidz627/fix/xfsUnmount

GetMountRefs fixed to handle corrupted mounts by treating it like an unmounted volume

roycaihw committed:
kube-aggregator: periodically resync local specs

Co-authored-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

k8s-ci-robot committed:
Merge pull request #74653 from cofyc/fix74650-CheckVolumeExistenceOperation

Distinguish between volume path and mount path

k8s-ci-robot committed:
Merge pull request #74688 from ksubrmnn/overlay_dsr

Allow Overlay loadbalancing to Public IP in Windows

roycaihw committed:
apiextensions: add openapi publishing unit+integration+e2e tests

Co-authored-by: Maciej Szulik <maszulik@redhat.com>
Co-authored-by: Dr. Stefan Schimanski <stefan.schimanski@gmail.com>

roycaihw committed:
generated

k8s-ci-robot committed:
Merge pull request #74713 from WanLinghao/duplicate_package

Remove duplicate package import

k8s-ci-robot committed:
Merge pull request #74738 from wk8/wk8/gmsa_e2e

Adding an e2e test on Windows GMSA support

k8s-ci-robot committed:
Merge pull request #74832 from roycaihw/kubectl-crd-test

kubectl e2e: test client-side validation behavior on CustomResources

k8s-ci-robot committed:
Merge pull request #74837 from mtaufen/godep-klog

Update klog dependency

k8s-ci-robot committed:
Merge pull request #74909 from NetApp/csi-raw-block-to-beta2

Promote CSI raw blocks to beta

k8s-ci-robot committed:
Merge pull request #71651 from Pingan2017/cronjob-add

add subcommand `kubectl create cronjob`

k8s-ci-robot committed:
Merge pull request #74936 from rjaini/rj_1.14

Adding a check to make sure Metadata is read only if flag value is true.

k8s-ci-robot committed:
Merge pull request #75055 from pjh/gce-windows-disable-defender-again

Disable Windows Defender in Windows nodes on GCE again.

k8s-ci-robot committed:
Merge pull request #75056 from timothysc/configuser

Revert "Merge pull request #74628 from hpandeycodeit/alpha_kubeconf"

k8s-ci-robot committed:
Merge pull request #75061 from andrewsykim/fix-mac-bazel

fix hack/update-bazel.sh for mac os environments

tnozicka committed:
Fix `kubectl run` watches in waitForPod

tnozicka committed:
Generated: Update Bazel

k8s-ci-robot committed:
Merge pull request #75064 from apelisse/fix-volume-test

Ignore changes to managed field in noderestriction

k8s-ci-robot committed:
Merge pull request #75067 from msau42/csi-beta-e2e

make CSI PD tests serial

k8s-ci-robot committed:
Merge pull request #71192 from roycaihw/crd-publish-openapi

Publish CRD openapi

k8s-ci-robot committed:
Merge pull request #75082 from neolit123/reset-clean-status

kubeadm-reset: add means to clear the ClusterStatus

k8s-ci-robot committed:
Merge pull request #74938 from brusMX/win-e2e-nodeport

Creating NodePort test for Windows

pohly committed:
e2e/storage: fix snapshot support in external driver testing

When using an already installed driver, the snapshot name is the
original driver name. Renaming was incorrectly copied from the in-tree
CSI hostpath driver.

adelina-t committed:
Update gmsa test

Wait for the container to be in Running state before running any command.

pohly committed:
e2e/storage: allow empty volume type

This can happen when the test pattern is about snapshotting.

dims committed:
Stop waiting on conformance image pod when it fails

Change-Id: I516833167f8e0aaa0e7962c1d6bdbe2ca422dc6e

k8s-ci-robot committed:
Merge pull request #75038 from soltysh/makefile_update

Add examples how to run make test-cmd specific tests

mbohlool committed:
Refactor CR converter code to share code between nopConverter and webhookConverter

gnufied committed:
Fix volume attach limit flake

It looks like node does become unschedulable for the pod
but condition does not get added to the pod in time.

Also ginkgo could retry the test and hence it helps to use
unique node label for scheduling.

gnufied committed:
Fix emptydir e2e flake

Wait forver and reduce grace period deletion

k8s-ci-robot committed:
Merge pull request #74904 from sttts/sttts-proto-tests

Rework proto generation scripts and add tests

mbohlool committed:
update generated files

mtaufen committed:
Windows kubeproxy flags cleanup

- Pass most of the flags via kube-env, similar to Kubelet, to make it
  easier to reconfigure.
- Use kube-proxy's kubeconfig file to pass the API server address,
  rather than the --master flag.
- Make KUBEPROXY_ARGS a required variable in Windows kube-env, similar
  to KUBELET_ARGS.

Random-Liu committed:
Update cadvisor to v0.33.1.

liggitt committed:
Ensure namespaces exist on start

mbohlool committed:
Add latency metric for CR Webhook Conversion

Commit:
use utilsets.NewString

Commit:
Feature-gate the APPLY metric value

dashpole committed:
add kubelet prometheus resource metrics endpoint

yguo0905 committed:
GCE: bump COS image version to cos-beta-73-11647-64-0

k8s-ci-robot committed:
Merge pull request #73383 from danielqsj/zkr

bump zookeeper-installer image version in e2e

k8s-ci-robot committed:
Merge pull request #74788 from feiskyer/windows-network-stats

Add network stats for Windows nodes and containers

k8s-ci-robot committed:
Merge pull request #74913 from yastij/explicit-restart-policy

explicit the restartPolicy on the dockershim

k8s-ci-robot committed:
Merge pull request #75121 from dims/stop-waiting-on-conformance-image-pod-failure

Stop waiting on conformance image pod when it fails

k8s-ci-robot committed:
Merge pull request #75124 from gnufied/fix-attach-limit-flake

Fix volume attach limit flake

Commit:
Generated files

pjh committed:
Don't test metadata server connectivity from GCE Windows pods.

Windows containers do not include a route to the GCE metadata server by
default. This check is causing the "DNS should provide DNS for the
cluster" test to fail for clusters with Windows nodes
(https://testgrid.k8s.io/sig-windows#gce-windows-master&width=20).

Tested that this works by running "DNS should provide DNS for the
cluster" against an e2e cluster with Windows nodes brought up on GCE.

Commit:
Fix version converter

Commit:
add test to verify fix

k8s-ci-robot committed:
Merge pull request #74835 from davidz627/feature/adcFallback

Add logic for initializing CSINode on Node startup for CSI Migration [Replaces #70909]

k8s-ci-robot committed:
Merge pull request #74433 from tallclair/runtimeclass-internal

Migrate RuntimeClass from a CRD to an internal API

k8s-ci-robot committed:
Merge pull request #74434 from freehan/pod-ready-ga

promote PodReadinessGate feature to GA

k8s-ci-robot committed:
Merge pull request #74442 from yujuhong/reconcile-node-labels

nodelifecycle controller: reconcile node OS/arch labels

k8s-ci-robot committed:
Merge pull request #74544 from gnufied/deprecate-cloudprovider-predicates

Deprecate cloudprovider specific volume limit predicates

k8s-ci-robot committed:
Merge pull request #74990 from jingxu97/pd_windows_test

Add GCE PD tests for windows cluster

k8s-ci-robot committed:
Merge pull request #73946 from dashpole/prometheus_core_metrics

Add kubelet resource metrics v1alpha1 endpoint

k8s-ci-robot committed:
Merge pull request #75014 from mtaufen/windows-logrotate

Windows log rotation

k8s-ci-robot committed:
Merge pull request #75023 from danielqsj/metrics-1

clean the deprecated metrics which introduced recently

k8s-ci-robot committed:
Merge pull request #75096 from tnozicka/fix-kubectl-run-watches

Fix kubectl run watches

k8s-ci-robot committed:
Merge pull request #75116 from adelina-t/patch-1

Update gmsa test

k8s-ci-robot committed:
Merge pull request #75139 from mtaufen/windows-kube-proxy-args

Windows kubeproxy flags cleanup

k8s-ci-robot committed:
Merge pull request #75140 from Random-Liu/cadvisor-0-33-1

Update cadvisor to v0.33.1.

k8s-ci-robot committed:
Merge pull request #75141 from liggitt/deflake-max_json_patch_operations_test

Ensure namespaces exist on start

k8s-ci-robot committed:
Merge pull request #75152 from pjh/gce-windows-dns-testfix

Don't test metadata server connectivity from GCE Windows pods.

k8s-ci-robot committed:
Merge pull request #74998 from mbohlool/pippin

Webhook configurations can choose which version of Review request they accept

mm4tt committed:
Disable NodeProblemDector test in large clusters.

It's suspected of causing timeouts in scalability 5k node cluster
tests, see https://github.com/kubernetes/kubernetes/issues/75095#issuecomment-470912583

k8s-ci-robot committed:
Merge pull request #75044 from errordeveloper/move-cli-runtime-subpackages

Move printers & resources packages

liggitt committed:
ignore uncommitted generated openapi files in verify-godeps

k8s-ci-robot committed:
Merge pull request #74863 from gnufied/csi-volume-expansion

CSI volume expansion

andrewsykim committed:
add label area/test in k8s.io/kubernetes/test/*

k8s-ci-robot committed:
Merge pull request #75131 from gnufied/increase-pod-wait-time

Fix emptydir e2e flake

joelsmith committed:
Update embargo doc link in SECURITY_OWNERS and changes PST to PSC

joelsmith committed:
Also update SECURITY_CONTACTS from staging

tallclair committed:
Update tests for RuntimeClass beta

k8s-ci-robot committed:
Merge pull request #74086 from vladimirvivien/csi-inline-volume-api

CSI Inline Volume Implementation

k8s-ci-robot committed:
Merge pull request #74376 from mbohlool/gimli

Add latency metric for CR Webhook Conversion

k8s-ci-robot committed:
Merge pull request #75063 from wangzhen127/npd-test-fix

Fix NPD e2e test on Ubuntu node and update NPD container version

k8s-ci-robot committed:
Merge pull request #75075 from kwiesmueller/fix-fieldmanager-sorting

fix handling of nil times in managedFields sorting

k8s-ci-robot committed:
Merge pull request #75149 from yguo0905/m73

GCE: bump COS image version to cos-beta-73-11647-64-0

k8s-ci-robot committed:
Merge pull request #75184 from liggitt/verify-godeps-generated

ignore uncommitted generated openapi files in verify-godeps

k8s-ci-robot committed:
Merge pull request #74997 from jennybuckley/apply-metrics

Track dry-run and apply in metrics

k8s-ci-robot committed:
Merge pull request #75048 from dims/add-verify-check-for-publish-bot-dependencies

verify script to ensure publishing bot rules and godeps are in sync

k8s-ci-robot committed:
Merge pull request #75192 from andrewsykim/area-test

add label area/test in k8s.io/kubernetes/test/*

k8s-ci-robot committed:
Merge pull request #75201 from joelsmith/secdoc

Update embargo doc link in SECURITY_OWNERS and change PST to PSC

k8s-ci-robot committed:
Merge pull request #75003 from tallclair/runtimeclass-beta

Promote RuntimeClass to beta

k8s-ci-robot committed:
Merge pull request #75073 from grayluck/esipp

Allow session affinity a period of time to setup for new services.

k8s-ci-robot committed:
Merge pull request #74441 from Random-Liu/pod-log-directory

Add namespace and name into the CRI pod log directory

k8s-ci-robot committed:
Merge pull request #74760 from apelisse/add-mangerfield-flag

Add "fieldManager" to flag to PATCH/CREATE/UPDATE

k8s-ci-robot committed:
Merge pull request #75177 from mm4tt/node-problem-detector

Disable NodeProblemDector test in large clusters.

k8s-ci-robot committed:
Merge pull request #75151 from jennybuckley/fix-conversion-bug

Fix apply conversion bug

k8s-ci-robot committed:
Merge pull request #75133 from pohly/storage-external-snapshot

e2e: external snapshotter

kairen committed:
Fix conflicting files
